scrapy/core/downloader/handlers/http11.py

```diff
@@ -9,16 +9,21 @@
 from urllib.parse import urldefrag, urlunparse
 
 from twisted.internet import defer, protocol, ssl
+from twisted.internet.defer import CancelledError, Deferred, fail, maybeDeferred
 from twisted.internet.endpoints import TCP4ClientEndpoint
 from twisted.internet.error import TimeoutError
 from twisted.python.failure import Failure
-from twisted.web.client import (
-    URI,
-    Agent,
-    HTTPConnectionPool,
-    ResponseDone,
-    ResponseFailed,
+from twisted.web._newclient import HTTP11ClientProtocol as TxHTTP11ClientProtocol
+from twisted.web._newclient import HTTPClientParser as TxHTTPClientParser
+from twisted.web._newclient import (
+    RequestGenerationFailed,
+    RequestNotSent,
+    TransportProxyProducer,
 )
+from twisted.web.client import URI, Agent
+from twisted.web.client import HTTPConnectionPool as TxHTTPConnectionPool
+from twisted.web.client import ResponseDone, ResponseFailed
+from twisted.web.client import _HTTP11ClientFactory as TxHTTP11ClientFactory
```

Comment:
I was planning on avoiding importing private stuff from Twisted, but things got quite complicated quickly. I think it may be easier to deal with breakages as they come.